lib/lte_link_control/lte_lc.c

```diff
@@ -199,6 +198,86 @@ static bool is_relevant_notif(const char *notif, enum lte_lc_notif_type *type)
 	return false;
 }
 
+static int parse_cereg(const char *notification,
+		       enum lte_lc_nw_reg_status *reg_status,
+		       struct lte_lc_cell *cell,
+		       struct lte_lc_psm_cfg *psm_cfg)
+{
+	int err, status;
+	struct at_param_list resp_list = {0};
+	char str_buf[10];
+	size_t len = ARRAY_SIZE(str_buf);
+
+	if (notification == NULL) {
+		return -EINVAL;
+	}
+
+	err = at_params_list_init(&resp_list, AT_CEREG_PARAMS_COUNT_MAX);
+	if (err) {
+		LOG_ERR("Could not init AT params list, error: %d", err);
+		return err;
+	}
+
+	/* Parse CEREG response and populate AT parameter list */
+	err = at_parser_max_params_from_str(notification,
+					    NULL,
+					    &resp_list,
+					    AT_CEREG_PARAMS_COUNT_MAX);
+	if (err) {
+		LOG_ERR("Could not parse AT+CEREG response, error: %d", err);
+		goto clean_exit;
+	}
+
+	/* Parse network registration status */
+	err = at_params_int_get(&resp_list,
+				AT_CEREG_REG_STATUS_INDEX,
+				&status);
+	if (err) {
+		LOG_ERR("Could not get registration status, error: %d", err);
+		goto clean_exit;
+	}
+
+	*reg_status = status;
+
+	/* Parse tracking area code */
+	err = at_params_string_get(&resp_list,
+				   AT_CEREG_TAC_INDEX,
+				   str_buf, &len);
+	if (err) {
+		LOG_ERR("Could not get tracking area code, error: %d", err);
+		goto clean_exit;
+	}
+
+	str_buf[len] = '\0';
```

Comment:
Theoretically, we should initialize `len` to `sizeof(str_buf) - 1`, otherwise, we might overwrite the last character in the string.